miden-tx/src/compiler/mod.rs

```diff
@@ -107,8 +109,8 @@ impl TransactionCompiler {
         &mut self,
         account_id: AccountId,
         notes: &[RecordedNote],
-        tx_script: Option<ProgramAst>,
-    ) -> Result<(Program, Option<Digest>), TransactionCompilerError> {
+        tx_script: &Option<TransactionScript>,
+    ) -> Result<Program, TransactionCompilerError> {
```

Comment:
For transaction compiler, we could probably keep the old interface and it doesn't really need the inputs to compile transaction.